internal/processing/status/status_test.go

```diff
@@ -60,6 +63,141 @@ var _ = Describe("Status", func() {
+				s := v2alpha1status.ReconciliationV2alpha1Status{
+					ApiRuleStatus: &gatewayv2alpha1.APIRuleStatus{},
+				}
+				failures := []validation.Failure{
+					{
+						AttributePath: "name",
+						Message:       "is wrong",
+					},
+					{
+						AttributePath: "service.name",
+						Message:       "is too short",
+					},
+				}
+				s.GenerateStatusFromFailures(failures)
+				Expect(s.ApiRuleStatus.State).To(Equal(gatewayv2alpha1.Error))
+				Expect(s.ApiRuleStatus.Description).To(Equal("Validation errors: Attribute 'name': is wrong\nAttribute 'service.name': is too short"))
+			})
+			It("should generate status for 5 failures", func() {
+				s := v2alpha1status.ReconciliationV2alpha1Status{
+					ApiRuleStatus: &gatewayv2alpha1.APIRuleStatus{},
+				}
+				failures := []validation.Failure{
+					{
+						AttributePath: "name",
+						Message:       "is wrong",
+					},
+					{
+						AttributePath: "service.name",
+						Message:       "is too short",
+					},
+					{
+						AttributePath: "service.port",
+						Message:       "is too big",
+					},
+					{
+						AttributePath: "service.host",
+						Message:       "is invalid",
+					},
+					{
+						AttributePath: "service.host",
+						Message:       "is too short",
+					},
+				}
+				s.GenerateStatusFromFailures(failures)
+				Expect(s.ApiRuleStatus.State).To(Equal(gatewayv2alpha1.Error))
+				Expect(s.ApiRuleStatus.Description).To(Equal("Validation errors: " +
+					"Attribute 'name': is wrong\n" +
+					"Attribute 'service.name': is too short\n" +
+					"Attribute 'service.port': is too big\n" +
+					"2 more error(s)..."))
+			})
+			It("should generate Ready for no failures", func() {
+				s := v2alpha1status.ReconciliationV2alpha1Status{
+					ApiRuleStatus: &gatewayv2alpha1.APIRuleStatus{},
+				}
+				s.GenerateStatusFromFailures(nil)
+				Expect(s.ApiRuleStatus.State).To(Equal(gatewayv2alpha1.Ready))
+				Expect(s.ApiRuleStatus.Description).To(Equal("No errors"))
+			})
+		})
+		Context("GetStatusForErrorMap", func() {
+			It("should set Error state for errorMap", func() {
+				s := v2alpha1status.ReconciliationV2alpha1Status{
+					ApiRuleStatus: &gatewayv2alpha1.APIRuleStatus{},
+				}
+				errMap := map[v2alpha1status.ResourceSelector][]error{
+					0: {errors.New("one error"), errors.New("another error")},
+					1: {errors.New("one error"), errors.New("another error")},
+				}
+				s.GetStatusForErrorMap(errMap)
+				Expect(s.ApiRuleStatus.State).To(Equal(gatewayv2alpha1.Error))
+				Expect(s.ApiRuleStatus.Description).To(Equal(
+					"ApiRuleErrors: one error, another error\n" +
+						"VirtualServiceErrors: one error, another error"))
+			})
+			It("should set Ready state for no errorMap", func() {
+				s := v2alpha1status.ReconciliationV2alpha1Status{
+					ApiRuleStatus: &gatewayv2alpha1.APIRuleStatus{},
+				}
+				s.GenerateStatusFromFailures(nil)
+				Expect(s.ApiRuleStatus.State).To(Equal(gatewayv2alpha1.Ready))
```

Comment:
test also if description is set back to empty / no errors, perhaps when previously was not.